xcode/providers.bzl

```diff
@@ -18,3 +18,89 @@ visibility("public")
 
 # TODO: b/311385128 - Migrate the native implementation here.
 XcodeVersionInfo = apple_common.XcodeVersionConfig
+
+XcodeSdkVariantInfo = provider(
+    doc = """\
+        Contains information about a specific SDK that is included in a version
+        of Xcode.
+        """,
+    fields = {
+        "archs": """\
+            `list[str]`. The CPU architectures that are valid target
+            architectures when building with this SDK.
+            """,
+        "build_version": """\
+            `str`. The alpha-numeric build version component of the SDK. This is
+            most often used to distinguish between different beta releases of
+            the same SDK.
+            """,
+        "clangrt_name": """\
+            `str`. The platform-specific component of the name of the clang-rt
+            built-in libraries that should be linked into binaries when using
+            features that require them, such as sanitizers.
+            """,
+        "device_families": """\
+            `dict[str, str]`. The device families that are supported when
+            compiling resources with this SDK. Each key in the dictionary is the
+            name of a device family as it appears on the command line of
+            resource processing tools, and the corresponding value is the
+            numeric identifier used when referencing the device family in the
+            `UIDeviceFamily` key of a bundle's `Info.plist` file.
+            """,
+        "llvm_triple_environment": """\
+            `str`. The environment component of the LLVM triple that is used to
+            compile code when building with this SDK. May be the empty string
+            for device builds or platforms that do not use an environment
+            component, like macOS.
+            """,
+        "llvm_triple_os": """\
+            `str`. The operating system component of the LLVM triple that is
+            used to compile code when building with this SDK.
+            """,
+        "llvm_triple_vendor": """\
+            `str`. The vendor component of the LLVM triple that is used to
+            compile code when building with this SDK.
+            """,
+        "maximum_supported_os_version": """\
+            `apple_common.dotted_version`. The highest operating system version
+            that is supported when building with this SDK.
+            """,
+        "minimum_supported_os_version": """\
```

Comment:
This will be good for **rules_swift_package_manager**.

Reply:
Yes! I actually hit this issue on a random side-project im working on targeting macOS 15. So im excited to be able to fix: https://github.com/cgrindel/rules_swift_package_manager/issues/892

Reply:
cc: @cgrindel